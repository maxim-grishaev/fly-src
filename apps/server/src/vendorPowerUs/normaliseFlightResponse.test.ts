import { S1 } from './fixture.mock';
import {
  normaliseTicket,
  normaliseFlightResponse,
} from './normaliseFlightResponse';

describe('normaliseResponse', () => {
  it('should normalise response', () => {
    const result = normaliseFlightResponse(S1, 60 * 60 * 1000);
    expect(result).toMatchSnapshot();
  });
});

const nowOrig = Date.now;
Date.now = () => 123;
afterAll(() => {
  Date.now = nowOrig;
});

describe('normaliseFlight', () => {
  it('should normalise the response', () => {
    const result = normaliseTicket(S1.flights[0], 60 * 60 * 1000);
    expect(result).toMatchInlineSnapshot(`
APITicket {
  "bestBefore": 1970-01-01T01:00:00.123Z,
  "cacheTTLMs": 3600000,
  "flights": [
    APITicketFlight {
      "flightDuration": 115,
      "flightNumber": "144",
      "fromPlace": "Schonefeld",
      "fromTime": 2019-08-08T04:30:00.000Z,
      "id": "9708aa1e09",
      "toPlace": "Stansted",
      "toTime": 2019-08-08T06:25:00.000Z,
    },
    APITicketFlight {
      "flightDuration": 120,
      "flightNumber": "8542",
      "fromPlace": "Stansted",
      "fromTime": 2019-08-10T05:35:00.000Z,
      "id": "18120bb1d2",
      "toPlace": "Schonefeld",
      "toTime": 2019-08-10T07:35:00.000Z,
    },
  ],
  "id": "7bc756ad54",
  "price": APIMonetary {
    "amount": "129.00",
    "currency": "EUR",
  },
  "vendorId": "powerUs",
}
`);
  });
});
