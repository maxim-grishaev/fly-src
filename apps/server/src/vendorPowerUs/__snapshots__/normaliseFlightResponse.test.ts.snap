// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normaliseResponse should normalise response 1`] = `
[
  APITicket {
    "bestBefore": 1970-01-01T01:00:00.123Z,
    "cacheTTLMs": 3600000,
    "flights": [
      APITicketFlight {
        "flightDuration": 115,
        "flightNumber": "144",
        "fromPlace": "Schonefeld",
        "fromTime": 2019-08-08T04:30:00.000Z,
        "id": "9708aa1e09",
        "toPlace": "Stansted",
        "toTime": 2019-08-08T06:25:00.000Z,
      },
      APITicketFlight {
        "flightDuration": 120,
        "flightNumber": "8542",
        "fromPlace": "Stansted",
        "fromTime": 2019-08-10T05:35:00.000Z,
        "id": "18120bb1d2",
        "toPlace": "Schonefeld",
        "toTime": 2019-08-10T07:35:00.000Z,
      },
    ],
    "id": "64616c2e89",
    "price": APIMonetary {
      "amount": "129.00",
      "currency": "EUR",
    },
    "vendorId": "powerUs",
  },
  APITicket {
    "bestBefore": 1970-01-01T01:00:00.123Z,
    "cacheTTLMs": 3600000,
    "flights": [
      APITicketFlight {
        "flightDuration": 120,
        "flightNumber": "8545",
        "fromPlace": "Schonefeld",
        "fromTime": 2019-08-08T20:25:00.000Z,
        "id": "8b6f9c5add",
        "toPlace": "Stansted",
        "toTime": 2019-08-08T22:25:00.000Z,
      },
      APITicketFlight {
        "flightDuration": 110,
        "flightNumber": "145",
        "fromPlace": "Stansted",
        "fromTime": 2019-08-10T06:50:00.000Z,
        "id": "aba024a07e",
        "toPlace": "Schonefeld",
        "toTime": 2019-08-10T08:40:00.000Z,
      },
    ],
    "id": "e472a80177",
    "price": APIMonetary {
      "amount": "134.81",
      "currency": "EUR",
    },
    "vendorId": "powerUs",
  },
  APITicket {
    "bestBefore": 1970-01-01T01:00:00.123Z,
    "cacheTTLMs": 3600000,
    "flights": [
      APITicketFlight {
        "flightDuration": 120,
        "flightNumber": "8545",
        "fromPlace": "Schonefeld",
        "fromTime": 2019-08-08T20:25:00.000Z,
        "id": "8b6f9c5add",
        "toPlace": "Stansted",
        "toTime": 2019-08-08T22:25:00.000Z,
      },
      APITicketFlight {
        "flightDuration": 120,
        "flightNumber": "8542",
        "fromPlace": "Stansted",
        "fromTime": 2019-08-10T05:35:00.000Z,
        "id": "18120bb1d2",
        "toPlace": "Schonefeld",
        "toTime": 2019-08-10T07:35:00.000Z,
      },
    ],
    "id": "bd06e8b544",
    "price": APIMonetary {
      "amount": "134.81",
      "currency": "EUR",
    },
    "vendorId": "powerUs",
  },
  APITicket {
    "bestBefore": 1970-01-01T01:00:00.123Z,
    "cacheTTLMs": 3600000,
    "flights": [
      APITicketFlight {
        "flightDuration": 115,
        "flightNumber": "146",
        "fromPlace": "Schonefeld",
        "fromTime": 2019-08-08T16:00:00.000Z,
        "id": "550a52736c",
        "toPlace": "Stansted",
        "toTime": 2019-08-08T17:55:00.000Z,
      },
      APITicketFlight {
        "flightDuration": 110,
        "flightNumber": "145",
        "fromPlace": "Stansted",
        "fromTime": 2019-08-10T06:50:00.000Z,
        "id": "aba024a07e",
        "toPlace": "Schonefeld",
        "toTime": 2019-08-10T08:40:00.000Z,
      },
    ],
    "id": "789bc2db9d",
    "price": APIMonetary {
      "amount": "147.90",
      "currency": "EUR",
    },
    "vendorId": "powerUs",
  },
  APITicket {
    "bestBefore": 1970-01-01T01:00:00.123Z,
    "cacheTTLMs": 3600000,
    "flights": [
      APITicketFlight {
        "flightDuration": 120,
        "flightNumber": "8543",
        "fromPlace": "Schonefeld",
        "fromTime": 2019-08-08T08:00:00.000Z,
        "id": "04bb7ca48a",
        "toPlace": "Stansted",
        "toTime": 2019-08-08T10:00:00.000Z,
      },
      APITicketFlight {
        "flightDuration": 110,
        "flightNumber": "145",
        "fromPlace": "Stansted",
        "fromTime": 2019-08-10T06:50:00.000Z,
        "id": "aba024a07e",
        "toPlace": "Schonefeld",
        "toTime": 2019-08-10T08:40:00.000Z,
      },
    ],
    "id": "dba770f15e",
    "price": APIMonetary {
      "amount": "147.90",
      "currency": "EUR",
    },
    "vendorId": "powerUs",
  },
]
`;
